### ISSUE TITLE
syntax error when parsing view ddl includes `select distinct on` on postgresql

### ISSUE THREAD

**Issue by sanemat** [sqldef/sqldef#320]
I tried to export existing view includes `select distinct on`, but I got syntax error during parsing DDL.

psqldef 0f29f9424 (v 0.14.2)

## --export output

```
2022/11/26 18:52:39 found syntax error when parsing DDL "CREATE VIEW public.hoge_view AS SELECT DISTINCT ON (hoge.amount) hoge.amount FROM hoge": syntax error at position 51 near 'on'
```

## Input SQL

```
create table hoge (
  amount  text
);
create view hoge_view as
select distinct on (amount) * from hoge;
```

**Comment by k0kubun** [comment]
Thank you for your report. Would you be interested in filing a pull request to fix this? For psqldef, we use PostgreSQL's original parser and then convert its AST to sqldef's AST in https://github.com/k0kubun/sqldef/blob/master/database/postgres/parser.go. It's likely we're missing the conversion for `select distinct on`.

**Comment by sanemat** [comment]
I added these into `database/postgres/tests.yml`:

```
CreateViewWithDistinct: |
  create view hoge_view as
  select distinct amount from hoge;
CreateViewWithDistinctOn: |
  create view hoge_view as
  select distinct on (amount) * from hoge;
```

```
$ go test ./database/postgres                          
--- FAIL: TestParse (0.00s)
    --- FAIL: TestParse/CreateViewWithDistinct (0.00s)
        parser_test.go:22: unhandled node in parseSelectStmt: &pg_query.SelectStmt{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(nil)}, sizeCache:0, unknownFields:[]uint8(nil), DistinctClause:[]*pg_query.Node{(*pg_query.Node)(0xc000033a40)}, IntoClause:(*pg_query.IntoClause)(nil), TargetList:[]*pg_query.Node{(*pg_query.Node)(0xc000033a80)}, FromClause:[]*pg_query.Node{(*pg_query.Node)(0xc000033b80)}, WhereClause:(*pg_query.Node)(nil), GroupClause:[]*pg_query.Node(nil), HavingClause:(*pg_query.Node)(nil), WindowClause:[]*pg_query.Node(nil), ValuesLists:[]*pg_query.Node(nil), SortClause:[]*pg_query.Node(nil), LimitOffset:(*pg_query.Node)(nil), LimitCount:(*pg_query.Node)(nil), LimitOption:1, LockingClause:[]*pg_query.Node(nil), WithClause:(*pg_query.WithClause)(nil), Op:1, All:false, Larg:(*pg_query.SelectStmt)(nil), Rarg:(*pg_query.SelectStmt)(nil)}
    --- FAIL: TestParse/CreateViewWithDistinctOn (0.00s)
        parser_test.go:22: unhandled node in parseSelectStmt: &pg_query.SelectStmt{state:impl.MessageState{NoUnkeyedLiterals:pragma.NoUnkeyedLiterals{}, DoNotCompare:pragma.DoNotCompare{}, DoNotCopy:pragma.DoNotCopy{}, atomicMessageInfo:(*impl.MessageInfo)(nil)}, sizeCache:0, unknownFields:[]uint8(nil), DistinctClause:[]*pg_query.Node{(*pg_query.Node)(0xc0000fd240)}, IntoClause:(*pg_query.IntoClause)(nil), TargetList:[]*pg_query.Node{(*pg_query.Node)(0xc0000fd300)}, FromClause:[]*pg_query.Node{(*pg_query.Node)(0xc0000fd3c0)}, WhereClause:(*pg_query.Node)(nil), GroupClause:[]*pg_query.Node(nil), HavingClause:(*pg_query.Node)(nil), WindowClause:[]*pg_query.Node(nil), ValuesLists:[]*pg_query.Node(nil), SortClause:[]*pg_query.Node(nil), LimitOffset:(*pg_query.Node)(nil), LimitCount:(*pg_query.Node)(nil), LimitOption:1, LockingClause:[]*pg_query.Node(nil), WithClause:(*pg_query.WithClause)(nil), Op:1, All:false, Larg:(*pg_query.SelectStmt)(nil), Rarg:(*pg_query.SelectStmt)(nil)}
FAIL
FAIL    github.com/k0kubun/sqldef/database/postgres     0.007s
FAIL
```

My understanding is the generic parser (parser/parser.go) parses CreateViewWithDistinct.
To parse `select distinct on`, we should handle stmt.DistinctClause in `func parseSelectStmt` on database/postgres/parser.go.
https://github.com/k0kubun/sqldef/blob/73a5d2b7991bd80e3393e222cd233661eecce806/database/postgres/parser.go#L188
Is this right?

**Comment by k0kubun** [comment]
Nice :+1: Yes, that sounds right. If you haven't tried it, https://github.com/k0kubun/pp might be useful for looking into the node.

**Comment by sanemat** [comment]
it worked already :+1:  thanks

refs: psqldef: Fix some bugs in v0.17.1 https://github.com/sqldef/sqldef/pull/498
